types/index.d.ts

```diff
@@ -1,4 +1,4 @@
-// Minimum TypeScript Version: 3.7
+// Minimum TypeScript Version: 4.2
```

Comment:
We are using variadic tuples with spread operator in the middle, typescript 4.2 is minimum for that.